### ISSUE TITLE
Guidance annotations are displaying in the wrong tab

### ISSUE THREAD

**Issue by sjDCC** [DMPRoadmap/roadmap#1409]
When an institution customises a funder template, they can add guidance to a specific question as annotations. This is displaying in the wrong tab. It shows up under the funder tab rather than in one for the institution

![customisation-guidance](https://user-images.githubusercontent.com/6169940/39180250-41c1e898-47ae-11e8-8cc6-b27a1097b890.PNG)

We discussed this on the call on 24th April and agreed a few actions:

- Guidance should be grouped together by owner. The first tab should pertain to the template owner (funder or org). Other tabs may refer to orgs or guidance groups e.g. maths and statistics

- Org tabs should combine annotations and themed guidance. Annotations should display above themed guidance. We may need to restrict the field size (i.e. introduce a scroll bar) to ensure this doesn't push themed guidance off the page as it displays opened out rather than in an accordion to click open

- We shouldn’t allow annotations to be turned off as they are part of the template creation (user is notified that they have created plan based on X template with customisations by X) but generic themed guidance by org can be switched off under plan details guidance config panel.

**Comment by jollopre** [comment]
In a attempt to clarify what has to be done on this ticket. @xsrust and @jollopre have been thinking to create a mockup to better understand what we want to display in the front end. Below the mockup:
![mockup](https://user-images.githubusercontent.com/7393044/40975217-1bedbd88-68c2-11e8-9ff7-b767334dfcd2.png)

- Each tab represents an Org which is determined by the guidance groups selected for a plan.
- Each tab content starts with the guidance by annotation, since it is more relevant to the current question followed by the guidance text (belonging to a guidance group selected) that matches the theme for the current question and org tab selected.
- First tab is omitted when a template is not a customization_of another.
- Second tab is always your org guidances (annotation or by theme)

**Comment by sjDCC** [comment]
This looks good - thanks @jollopre @xsrust We should definitely group all the guidance (annotations & themed) per org. I would be inclined not to write 'guidance by annotation' on the front end though as it's a bit jargony. Perhaps 'guidance by question' or 'specific guidance' or 'customisations' to differentiate it from the themes?

Re: "First tab is omitted when a template is not a customization_of another." I'm not sure this would be the case. There would still be guidance from the template owner (funder or org) even if it hasn't been customised

**Comment by stephaniesimms** [comment]
I agree w/ @sjDCC that end users don't need or want to understand the different methods for admins to provide guidance. I think something like 'Guidance by question' is the best language to differentiate and signal to end users that a particular piece of guidance is specific to that question/template.

**Comment by stephaniesimms** [comment]
the guidance is scrambled across all tabs and display areas now. based on the screenshots below here is what i think is happening: 
- example answers for the template and Org customization seem to be displaying in the appropriate location below the question and editor
- but the template example answer is duplicated as the first piece of "NSF Guidance" in the panel on the right. the only thing that should appear in this panel is the second paragraph of guidance text and it should not have the label "NSF Guidance"
- the Org customized answer is also duplicated in the panel on the right when I tab to UCOP as the first piece of "UCOP Guidance." the only thing that should appear in this panel is the second entry of guidance text with the label "Question-Level Guidance"
- if a question is customized by an Org the themed guidance does not appear. this question in the screenshot should also have themed guidance for Metadata & documentation in an accordion.
- the labels for themed guidance should not contain the Org name (in screenshot 3 "DMPTool" replaces UCOP in the database) since it already appears in the tab and is redundant here. the accordions should only have the theme name.

![screen shot 2018-06-15 at 3 49 33 pm](https://user-images.githubusercontent.com/12703638/41492520-7020dfbc-70b4-11e8-8cbd-b372f84c6230.png)


![screen shot 2018-06-15 at 3 53 22 pm](https://user-images.githubusercontent.com/12703638/41492522-74a8dd78-70b4-11e8-969a-28132a7fa04e.png)


![screen shot 2018-06-15 at 4 07 30 pm](https://user-images.githubusercontent.com/12703638/41492788-4f85e99e-70b6-11e8-9f03-4c2b00f578d9.png)

**Comment by stephaniesimms** [comment]
things are looking much better @xsrust - the one major issue i noticed is that i can't see a combination of "Question Specific Guidance" and themed guidance for any specific Org. So for example, in the screenshot below, the question is tagged with the theme "Data volume" and there should be an accordion with that theme label beneath the Question Specific Guidance in the UCSF tab. Based on my guidance selections, there should also be a UCOP tab for this question with the same Data volume themed guidance. It looks like every time I add Question Specific Guidance as a customization it replaces all of the themed guidance for all Orgs.

![screen shot 2018-07-02 at 10 39 17 am](https://user-images.githubusercontent.com/12703638/42178358-f4943342-7de4-11e8-99fb-10ebbcb973bc.png)

other minor issues:
- Question Specific Guidance sounds awkward to me, also needs a hyphen. I suggest using "Question-Level Guidance" - thoughts @sjDCC ? something even shorter would be better...
- re your question @sjDCC - when there is no guidance to display I prefer keeping the current behavior where we just display the Comments area. This will help users find any existing comments, in particular. Alternatively, we need to add a message that there is no guidance (see screenshot below with blank area that users will likely just skip over and miss the Comments).

![screen shot 2018-07-02 at 10 22 23 am](https://user-images.githubusercontent.com/12703638/42178589-da46d0ca-7de5-11e8-9da7-a1905f1425f2.png)

**Comment by xsrust** [comment]
@stephaniesimms 
That's funny because I seem to get both the annotational guidance and the themed guidance to display.  Are you sure the correct guidance groups are selected?

![screen shot 2018-07-03 at 10 55 44 am](https://user-images.githubusercontent.com/7352616/42213085-ce1de53e-7eaf-11e8-98cd-67c0bfd47971.png)

I agree Question Specific Guidance sounds awkward.  I cant come up with anything shorter than Question-Level Guidance though.  Perhaps we can brainstorm in today's meeting.

I've updated the behavior around comments

**Comment by stephaniesimms** [comment]
I'm probably just a bad tester. Your screenshot above looks good @xsrust - we'll test again to confirm that all guidance is displaying where it should.

One more adjustment: please try removing the label for "Question Specific Guidance" altogether.

**Comment by xsrust** [comment]
The label should be removed in the last PR I sent, I managed to update it during the meeting.

**Comment by sjDCC** [comment]
I've done some testing on this and may have uncovered an issue which explains why @stephaniesimms couldn't get question-specific guidance and themes together. It seems that when you customise a funder template (I worked on Arnold Foundation and DCC default for American Uni), the act of adding an annotation clears out the themes tagged on that question. @xsrust could you investigate what's going wrong on save? Otherwise all looks good.

![archiving-section](https://user-images.githubusercontent.com/6169940/42314165-57dcc1ae-803c-11e8-856e-a522bf2a5ff8.PNG)
![themes-removed](https://user-images.githubusercontent.com/6169940/42314168-5bf7cd60-803c-11e8-8169-d366ee86a59e.PNG)

**Comment by xsrust** [comment]
@sjDCC  @stephaniesimms 

The code in this ticket has nothing to do with the editing/saving of annotations so I'd like to break that off into a separate bug ticket.

For UAT, the only changes which have been made here are on the display for the plans page, and the template preview.

**Comment by sjDCC** [comment]
Fine by me. I assume you can replicate this issue then? Certainly one to fix in this sprint if do.

**Comment by xsrust** [comment]
Testing now and will create a bug ticket if I can replicate

**Comment by briri** [comment]
This may be due to the size of the browser window. We've noticed in some situations that the borders of the Tinymce editors disappear at various sizes. I would bet that this cropping happens in similar situations.

**Comment by sjDCC** [comment]
Display all working fine for me aside from #1684. Closing out, but please check too @stephaniesimms  Thanks @xsrust

**Comment by stephaniesimms** [comment]
looks great. everything in its place.

**Comment by stephaniesimms** [comment]
reopening this issue as I noticed a display problem w/guidance subgroups that may be related. See new issue #1738 for details.

**Comment by sjDCC** [comment]
Agree - split looks good
